contracts/BNUGToken.sol

```diff
@@ -0,0 +1,66 @@
+pragma solidity >=0.4.22 <0.9.0;
+
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+
+
+contract BNUGToken is ERC20Burnable {
```

Comment:
what you did here is you created 2 tokens. The one from OZ and the second ones you integrated as functions. OZ already solves that for you.